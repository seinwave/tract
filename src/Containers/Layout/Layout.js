import React from "react";
import SideBar from "../SideBar/SideBar";
import TopBar from "../TopBar/TopBar";
import { Container, Row, Col } from "react-bootstrap";
import FinancialPage from "../../Components/Pages/FinancialPage/FinancialPage";
import TeamPage from "../../Components/Pages/TeamPage/TeamPage";
import TicketPage from "../../Components/Pages/TicketPage/TicketPage";
import ProjectPage from "../../Components/Pages/ProjectPage/ProjectPage";
import Dashboard from "../../Components/Pages/Dashboard/Dashboard";

import {
  BrowserRouter as Router,
  Route,
  NavLink,
  Switch,
} from "react-router-dom";
import "./Layout.css";

function Layout() {
  return (
    <Router>
      <Container fluid>
        <Row className="topbar-row">
          <Col xs={12} id="topbar-wrapper">
            <TopBar />
          </Col>
        </Row>
        <Row>
          <Col className="sidebar-wrapper-col" id="sidebar-wrapper">
            <SideBar>
              <NavLink
                className="nav-item"
                to="/home"
                activeStyle={{ color: "red", fontWeight: "bold" }}
              >
                {" "}
               
                  <span className="nav-title">Home </span><span className="fa fa-home menu-icon"></span>
              </NavLink>

              <NavLink
                className="nav-item"
                to="/financials"
                activeStyle={{ color: "red", fontWeight: "bold" }}
              >
                <span className="nav-title">Financials </span><span className="fa fa-dollar menu-icon"></span>
              </NavLink>
              <NavLink
                className="nav-item"
                to="/projects"
                activeStyle={{ color: "red", fontWeight: "bold" }}
              >
                Projects <i className="fa fa-briefcase"></i>
              </NavLink>
              <NavLink
                className="nav-item"
                to="/tickets"
                activeStyle={{ color: "red", fontWeight: "bold" }}
              >
                Tickets <i className="fa fa-file"></i>
              </NavLink>
              <NavLink
                className="nav-item"
                to="/team"
                activeStyle={{ color: "red", fontWeight: "bold" }}
              >
                Team <i className="fa fa-cog"></i>
              </NavLink>
              <div class="h-divider"></div>
              <NavLink
                className="nav-item"
                to="/profile"
                activeStyle={{ color: "red", fontWeight: "bold" }}
              >
                Profile <i className="fa fa-user"></i>
              </NavLink>
              <NavLink
                className="nav-item"
                to="/exit"
                activeStyle={{ color: "red", fontWeight: "bold" }}
              >
                Sign Out <i className="fa fa-sign-out"></i>
              </NavLink>
            </SideBar>
          </Col>
          <Col xs={8} id="page-content-wrapper">
            <Switch>
              <Route path="/financials">
                <FinancialPage />
              </Route>
              <Route path="/home">
                <Dashboard />
              </Route>
              <Route path="/projects">
                <ProjectPage />
              </Route>
              <Route path="/team">
                <TeamPage />
              </Route>
              <Route path="/tickets">
                <TicketPage />
              </Route>
            </Switch>
          </Col>
        </Row>
      </Container>
    </Router>
  );
}

export default Layout;
