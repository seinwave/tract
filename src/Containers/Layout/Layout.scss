#page-content-wrapper {
    padding: 1rem;
    margin: 1rem;
}

.fa {
    text-align: right;
    padding-left: 2rem;
}

.menu-icon{
    text-align: center;
    position: absolute;
    right: 14px;
    width: 20px;
    top: 4px;
    margin: 0 !important;
    line-height: 1 !important;
    font-size: 1.13rem;
    transition: color .3s cubic-bezier(.645,.045,.355,1);
}

.nav-row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: stretch;
}

.topbar-wrapper {
    background-color: white;
}

.h-divider {
    margin-top: 5px;
    margin-bottom: 5px;
    height: 1px;
    width: 100%;
    border-top: 1px solid gray;
}

.body-row {
    background: #f3f3f3;
}