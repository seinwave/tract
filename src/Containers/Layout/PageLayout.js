import React, { useState } from "react";
import SideBar from "../SideBar/SideBar";
import TopBar from "../TopBar/TopBar";
import FinancialPage from "../../Components/Pages/FinancialPage/FinancialPage";
import TeamPage from "../../Components/Pages/TeamPage/TeamPage";
import TicketPage from "../../Components/Pages/TicketPage/TicketPage";
import ProjectPage from "../../Components/Pages/ProjectPage/ProjectPage";
import Dashboard from "../../Components/Pages/Dashboard/Dashboard";
import { Layout, Menu, Breadcrumb } from "antd";
import {
  BrowserRouter as Router,
  Route,
  NavLink,
  Switch,
} from "react-router-dom";
import {
  PieChartOutlined,
  SettingOutlined,
  DollarOutlined,
  InboxOutlined,
  TeamOutlined,
  BookOutlined,
  UserOutlined,
} from "@ant-design/icons";

import "./PageLayout.scss";
import "antd/dist/antd.css";

const { Header, Footer, Sider, Content } = Layout;
const { SubMenu } = Menu;

function PageLayout() {
  const [collapsed, setCollapsed] = useState(false);

  const onCollapse = () => {
    setCollapsed(!collapsed);
  };

  return (
    <>
      <Router>
        <Layout theme="light" hasSider={true}>
          <Sider
            theme="light"
            collapsible
            collapsed={collapsed}
            onCollapse={onCollapse}
            style={{
              overflow: "auto",
              height: "100vh",
              position: "fixed",
              left: 0,
            }}
          >
            <div className="logoContainer">
              <div className="logo">
                <img
                  src="src/images/logo192.png"
                  className="mr-2"
                  alt="Clean UI"
                />
                <div className="name">Tract</div>
              </div>
            </div>
            <Menu defaultSelectedKeys={["1"]} mode="inline">
              <Menu.Item key="1" icon={<PieChartOutlined />}>
                <NavLink className="nav-item" to="/home">
                  Home
                </NavLink>
              </Menu.Item>
              <Menu.Item key="2" icon={<TeamOutlined />}>
                <NavLink className="nav-item" to="/team">
                  Team
                </NavLink>
              </Menu.Item>
              <Menu.Item key="3" icon={<BookOutlined />}>
                <NavLink className="nav-item" to="/projects">
                  Projects
                </NavLink>
              </Menu.Item>
              <Menu.Item key="4" icon={<InboxOutlined />}>
                <NavLink className="nav-item" to="/tickets">
                  Tickets
                </NavLink>
              </Menu.Item>
              <Menu.Item key="5" icon={<DollarOutlined />}>
                <NavLink className="nav-item" to="/financials">
                  Financials
                </NavLink>
              </Menu.Item>

              <div class="h-divider"></div>
              <Menu.Item key="6" icon={<UserOutlined />}>
                <NavLink className="nav-item" to="/profile">
                  Profile
                </NavLink>
              </Menu.Item>
              <Menu.Item key="7" icon={<SettingOutlined />}>
                <NavLink className="nav-item" to="/exit">
                  Sign Out
                </NavLink>
              </Menu.Item>
            </Menu>
          </Sider>
          <Layout theme="light">
            <Header id="real-header" theme="light">
              <TopBar />
            </Header>
            <Content>
              <Switch>
                <Route path="/financials">
                  <FinancialPage />
                </Route>
                <Route path="/home">
                  <Dashboard />
                </Route>
                <Route path="/projects">
                  <ProjectPage />
                </Route>
                <Route path="/team">
                  <TeamPage />
                </Route>
                <Route path="/tickets">
                  <TicketPage />
                </Route>
              </Switch>
            </Content>
          </Layout>
        </Layout>
      </Router>
    </>
  );
}

export default PageLayout;
