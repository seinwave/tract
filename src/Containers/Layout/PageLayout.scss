#real-header {
    background: #fff;
    padding: 0;
    height: 87px; 
}

.logoContainer {
    height: 84px;
}

.logo {
    padding: 15px;
}
.corner-logo {
    height: 50px;
}

.name {
    font-weight: bold;
    padding-left: 15px;
    font-size: 20px;
    position: absolute;
    top: 25px;
}

// #page-content-wrapper {
//     padding: 1rem;
//     margin: 1rem;
// }

// .fa {
//     text-align: right;
//     padding-left: 2rem;
// }

// .menu-icon{
//     text-align: center;
//     position: absolute;
//     right: 14px;
//     width: 20px;
//     top: 4px;
//     margin: 0 !important;
//     line-height: 1 !important;
//     font-size: 1.13rem;
//     transition: color .3s cubic-bezier(.645,.045,.355,1);
// }

// .nav-row {
//     display: flex;
//     flex-direction: row;
//     justify-content: space-between;
//     align-items: stretch;
// }

// .topbar-wrapper {
//     background-color: white;
// }

.h-divider {
    margin-top: 5px;
    margin-bottom: 5px;
    margin-left: 15%;
    height: 1px;
    width: 70%;
    border-top: 1px solid gray;
}

