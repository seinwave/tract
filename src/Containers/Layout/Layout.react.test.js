import React from 'react';
import Layout from './Layout'
import renderer from 'react-test-renderer'

it('renders as expected', () => {
    const tree = renderer
    .create(<Layout/>)
    .toJSON();
    expect(tree).toMatchSnapshot(); 
})