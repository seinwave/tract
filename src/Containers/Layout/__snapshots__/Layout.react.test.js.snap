// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="container-fluid"
>
  <div
    className="topbar-row row"
  >
    <div
      className="col-12"
      id="topbar-wrapper"
    >
      <div
        className="topbar"
      >
        <nav
          className="navbar navbar-expand-lg navbar-light"
        >
          <img
            alt="tract-logo"
            className="topbar-logo-image"
            src="https://1000logos.net/wp-content/uploads/2017/11/Chipotle-Logo.png"
          />
          <a
            className="navbar-brand"
            href="www.mattseidholz.com/"
          >
            Tract
          </a>
          <div
            className="collapse navbar-collapse"
            id="navbarSupportedContent"
          />
          <div
            className="avatar-dropdown-button dropdown"
            onKeyDown={[Function]}
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <img
                alt="Avatarimage"
                className="topbar-avatar-image"
                src="https://cdn.iconscout.com/icon/free/png-256/avatar-375-456327.png"
              />
               
              Hi, USER
            </button>
          </div>
        </nav>
      </div>
    </div>
  </div>
  <div
    className="body-row row"
  >
    <div
      className="sidebar-wrapper-col col"
      id="sidebar-wrapper"
    >
      <div
        className="sidebar flex-column nav"
        onKeyDown={[Function]}
      >
        <div
          className="navlink-span"
        >
          <a
            aria-current={null}
            className="nav-item"
            href="/home"
            onClick={[Function]}
          >
             
            <span
              className="nav-title"
            >
              Home 
            </span>
            <span
              className="fa fa-home menu-icon"
            />
          </a>
        </div>
        <div
          className="navlink-span"
        >
          <a
            aria-current={null}
            className="nav-item"
            href="/team"
            onClick={[Function]}
          >
            Team 
            <i
              className="fa fa-cog"
            />
          </a>
        </div>
        <div
          className="navlink-span"
        >
          <a
            aria-current={null}
            className="nav-item"
            href="/projects"
            onClick={[Function]}
          >
            Projects 
            <i
              className="fa fa-briefcase"
            />
          </a>
        </div>
        <div
          className="navlink-span"
        >
          <a
            aria-current={null}
            className="nav-item"
            href="/tickets"
            onClick={[Function]}
          >
            Tickets 
            <i
              className="fa fa-file"
            />
          </a>
        </div>
        <div
          className="navlink-span"
        >
          <a
            aria-current={null}
            className="nav-item"
            href="/financials"
            onClick={[Function]}
          >
            <span
              className="nav-title"
            >
              Financials 
            </span>
            <span
              className="fa fa-dollar menu-icon"
            />
          </a>
        </div>
        <div
          className="navlink-span"
        >
          <div
            class="h-divider"
          />
        </div>
        <div
          className="navlink-span"
        >
          <a
            aria-current={null}
            className="nav-item"
            href="/profile"
            onClick={[Function]}
          >
            Profile 
            <i
              className="fa fa-user"
            />
          </a>
        </div>
        <div
          className="navlink-span"
        >
          <a
            aria-current={null}
            className="nav-item"
            href="/exit"
            onClick={[Function]}
          >
            Sign Out 
            <i
              className="fa fa-sign-out"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      className="col-8"
      id="page-content-wrapper"
    />
  </div>
</div>
`;
